front/src/actions/message.js

```diff
@@ -43,13 +43,17 @@ function createActions(store) {
       });
     },
     pushMessage(state, message) {
+      let newMessages = store.getState().messages;
+      if (message.uuid && message.uuid === newMessages[newMessages.length - 1].uuid) {
```

Comment:
This won't necessarily work, because if you sent 2 messages in a row, and the websocket event comes back after the 2nd message is received, then one message will be displayed twice.

I think we have no choice to deduplicate on the whole list of messages, or maybe we could keep a store of messages already displayed (like a javascript Map() )  ?

Reply:
Fixed by using filter on all messages.